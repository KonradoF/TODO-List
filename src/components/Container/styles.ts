import EStyleSheet from 'react-native-extended-stylesheet';

export const styles = EStyleSheet.create({
  wrapper: {
    backgroundColor: '$white',
    flex: 1,
    marginHorizontal: '$marginHorizontal',
  },
});
