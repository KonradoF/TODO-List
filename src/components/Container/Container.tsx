import React, {ReactElement, ReactNode} from 'react';
import {View, ViewStyle, SafeAreaView} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import {styles} from './styles';
import {Form} from 'src/components/Form/Form';

interface Props {
  shouldSafeAreaView?: boolean;
  keyboardAvoidingView?: boolean;
  children?: ReactNode;
  style?: ViewStyle;
  transparent?: boolean;
}

export const Container = ({
  shouldSafeAreaView = true,
  keyboardAvoidingView,
  children,
  style,
  transparent = false,
}: Props): ReactElement => {
  if (keyboardAvoidingView) {
    return (
      <Form>
        {shouldSafeAreaView ? (
          <SafeAreaView
            style={EStyleSheet.flatten([
              styles.wrapper,
              transparent && {backgroundColor: 'transparent'},
              style,
            ])}>
            {children}
          </SafeAreaView>
        ) : (
          <View
            style={EStyleSheet.flatten([
              styles.wrapper,
              transparent && {backgroundColor: 'transparent'},
              style,
            ])}>
            {children}
          </View>
        )}
      </Form>
    );
  }
  return shouldSafeAreaView ? (
    <SafeAreaView
      style={EStyleSheet.flatten([
        styles.wrapper,
        transparent && {backgroundColor: 'transparent'},
        style,
      ])}>
      {children}
    </SafeAreaView>
  ) : (
    <View
      style={EStyleSheet.flatten([
        styles.wrapper,
        transparent && {backgroundColor: 'transparent'},
        style,
      ])}>
      {children}
    </View>
  );
};
