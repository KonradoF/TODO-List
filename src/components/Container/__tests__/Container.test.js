import React from 'react';
import {shallow} from 'enzyme';
import {Container} from 'src/components/Container/Container';
import {Text} from 'react-native-elements';

describe('Container', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Container />);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - noSafeAreaView', () => {
      const component = shallow(<Container shouldSafeAreaView={false} />);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - keyboardAvoidingView', () => {
      const component = shallow(<Container keyboardAvoidingView />);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot  - shouldSafeAreaView && keyboardAvoidingView', () => {
      const component = shallow(
        <Container shouldSafeAreaView={false} keyboardAvoidingView />,
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe('Check render props', () => {
    let component;
    const text = 'Test';

    beforeEach(() => {
      component = shallow(
        <Container>
          <Text>{text}</Text>
        </Container>,
      );
      jest.clearAllMocks();
    });

    it('should render correct children', () => {
      const labelElement = component.find({children: text});
      expect(labelElement).toHaveLength(1);
    });
  });
  describe('Check render props noSafeAreaView', () => {
    let component;
    const text = 'Test';

    beforeEach(() => {
      component = shallow(
        <Container noSafeAreaView>
          <Text>{text}</Text>
        </Container>,
      );
      jest.clearAllMocks();
    });

    it('should render correct children', () => {
      const labelElement = component.find({children: text});
      expect(labelElement).toHaveLength(1);
    });
  });
  describe('Check render props keyboardAvoidingView', () => {
    let component;
    const text = 'Test';

    beforeEach(() => {
      component = shallow(
        <Container keyboardAvoidingView>
          <Text>{text}</Text>
        </Container>,
      );
      jest.clearAllMocks();
    });

    it('should render correct children', () => {
      const labelElement = component.find({children: text});
      expect(labelElement).toHaveLength(1);
    });
  });
  describe('Check render props keyboardAvoidingView noSafeAreaView', () => {
    let component;
    const text = 'Test';

    beforeEach(() => {
      component = shallow(
        <Container keyboardAvoidingView noSafeAreaView>
          <Text>{text}</Text>
        </Container>,
      );
      jest.clearAllMocks();
    });

    it('should render correct children', () => {
      const labelElement = component.find({children: text});
      expect(labelElement).toHaveLength(1);
    });
  });
});
