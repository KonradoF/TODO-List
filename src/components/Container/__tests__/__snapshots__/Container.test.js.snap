// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Rendering should match to snapshot  - shouldSafeAreaView && keyboardAvoidingView 1`] = `
<Form>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "marginHorizontal": 50,
      }
    }
  />
</Form>
`;

exports[`Container Rendering should match to snapshot - keyboardAvoidingView 1`] = `
<Form>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "marginHorizontal": 50,
      }
    }
  />
</Form>
`;

exports[`Container Rendering should match to snapshot - noSafeAreaView 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "marginHorizontal": 50,
    }
  }
/>
`;

exports[`Container Rendering should match to snapshot 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "marginHorizontal": 50,
    }
  }
/>
`;
